### ISSUE TITLE
MKtlLookup keeps adding new devices upon repeated initialisations

### ISSUE THREAD

**Issue by sensestage** [ModalityTeam/Modality-toolkit#220]
```
MKtl.find;

MKtlLookup.all[ 'midi_3_supercollider' ].destDevice
MKtlLookup.all[ 'midi_3_supercollider' ].srcDevice

-> [ MIDIEndPoint("SuperCollider", "in0"), MIDIEndPoint("SuperCollider", "in1"), MIDIEndPoint("SuperCollider", "in2") ]
-> MIDIEndPoint("SuperCollider", "out0")

MKtl.find;

MKtlLookup.all[ 'midi_3_supercollider' ].destDevice
MKtlLookup.all[ 'midi_3_supercollider' ].srcDevice

-> [ MIDIEndPoint("SuperCollider", "out0"), MIDIEndPoint("SuperCollider", "out0") ]
-> [ MIDIEndPoint("SuperCollider", "in0"), MIDIEndPoint("SuperCollider", "in1"), MIDIEndPoint("SuperCollider", "in2"), MIDIEndPoint("SuperCollider", "in0"), MIDIEndPoint("SuperCollider", "in1"), MIDIEndPoint("SuperCollider", "in2") ]
```

**Comment by adcxyz** [comment]
MIDIEndPoint now checks for equality, which should avoid multiple entries after rescanning.  
Marije, please check on linux! 
(still using arrays for now, need to think about simplest way to handle them)

**Comment by adcxyz** [comment]
cleaned up some more details there, so should work fine now AFAICT. 
please test on linux.

**Comment by LFSaw** [comment]
@sensestage ?

**Comment by sensestage** [comment]
with one small fix: 14388b9151f5d1eb6a212bd728f647c9e2649d81
it works now.
